test/list/any.wren

```diff
@@ -0,0 +1,10 @@
+var a = [1, 2, 3]
+
+var b = a.any {|x| x > 3 }
+IO.print(b) // expect: false
+
+var d = a.any {|x| x > 1 }
+IO.print(d) // expect: true
+
+var e = [].any {|x| true }
+IO.print(e) // expect: false
```

Comment:
Great tests!

Can you also add a test that it throws a runtime error if the argument isn't callable? You can just copy the `test/list/all_non_function_arg.wren` and change it to call `.any`.